### PR TITLE
ddl: resolve the charset by the order: table->database->server (#9105)

### PR DESCRIPTION
**Pull request by winkyao**
### What problem does this PR solve? 

cherry-pick from #9105

### CODE REVIEW

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 
 	. "github.com/pingcap/check"
 	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/model"
```
Comment (ciscoxll) [inline ddl/db_integration_test.go]: Delete this line?

Comment (winkyao) [reply ddl/db_integration_test.go]: ?

**Comment by zimulala** [comment]
/run-all-tests tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1

**Comment by winkyao** [comment]
/run-all-tests tidb-test=pr/730 tidb-private-test=release-2.1 tikv=release-2.1

**Comment by winkyao** [comment]
/run-all-tests tidb-test=pr/730 tidb-private-test=release-2.1 tikv=release-2.1

**Pull request merged**